src/views/PositionPdfPlot/PositionPdfPlotWidget.tsx

```diff
@@ -70,103 +71,98 @@ const PositionPdfPlotWidget: FunctionComponent<Props> = ({fetchSegment, startTim
-                    const v = (vv - minValue) / (maxValue - minValue)
-                    const col = colorForValue(v)
-                    imageData.data[i + 0] = col[0]
-                    imageData.data[i + 1] = col[1]
-                    imageData.data[i + 2] = col[2]
-                    imageData.data[i + 3] = 255
-                }
-                else {
-                    imageData.data[i + 0] = 0
-                    imageData.data[i + 1] = 0
-                    imageData.data[i + 2] = 0
-                    imageData.data[i + 3] = 0
-                }
-                i += 4
+        const totalTimePoints = visibleValues.length
+        const totalLinearPositionBuckets = visibleValues[0].length
+        const data = []
+        const colorForValue = getColorForValueFn(minValue, maxValue)
+        // We need to do a non-obvious transpose on the input data, because the input
+        // is structured as [timeCode][linearPositionBucket] -- i.e. the left-most array index
+        // describes the time point, which should be the x-axis on the visualization.
+        // Unfortunately, this corresponds to a column-major order, while our graphics need
+        // to be described as a one-dimensional array in row-major order (i.e. subsequent entries
+        // move across rows from left to right), and we also need to invert the y-axis. So
+        // we do a manual double loop here instead of just a flatmap().
+        for (let positionIndex = totalLinearPositionBuckets - 1; positionIndex >= 0; positionIndex--) {
+            for (let timeIndex = 0; timeIndex < totalTimePoints; timeIndex++) {
+                const value = visibleValues[timeIndex][positionIndex]
+                data.push(colorForValue(value))
             }
         }
+        const clampedData = Uint8ClampedArray.from(data.flat())
+        const imageData = new ImageData(clampedData, totalTimePoints)
         return imageData
     }, [visibleValues, minValue, maxValue])
+    
+    const margins = useTimeseriesMargins(timeseriesLayoutOpts)
+    const adjustedHeight = linearPositions ? height - 50 : height // leave an additional margin for the checkbox if we have linear positions to display
+    const panelCount = 1
+    const toolbarWidth = timeseriesLayoutOpts?.hideToolbar ? 0 : DefaultToolbarWidth
+    const { panelWidth, panelHeight } = usePanelDimensions(width - toolbarWidth, adjustedHeight, panelCount, panelSpacing, margins)
+    
+    const canvas = useMemo(() => {
+        return document.createElement('canvas')
+    }, [])
 
     const paintPanel = useCallback((context: CanvasRenderingContext2D, props: PanelProps) => {
+        const canvas = props.offscreenCanvas
+        if (canvas === undefined) return
         if (!imageData) return
-        // Draw scaled version of image
-        // See: https://stackoverflow.com/questions/3448347/how-to-scale-an-imagedata-in-html-canvas
-        const canvas = document.createElement('canvas')
```

Comment:
I wasn't able to confirm, but I believe this was adding a new canvas to the document with every paint call. I memoized it out instead.